lib/elixir/lib/uri.ex

```diff
@@ -204,9 +208,9 @@ defmodule URI do
       true
 
   """
-  @spec char_reserved?(term) :: boolean
-  def char_reserved?(c) do
-    c in ':/?#[]@!$&\'()*+,;='
+  @spec char_reserved?(pos_integer) :: boolean
+  def char_reserved?(char) when is_integer(char) and char > 0 do
+    char in ':/?#[]@!$&\'()*+,;='
   end
```

Comment:
you are unnecessarily checking for `is_integer(char) and char > 0`. all that is implied by the `in` operator